src/Mvc/src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/BindingInfo.cs

```diff
@@ -48,9 +52,29 @@ public BindingInfo(BindingInfo other)
         public string BinderModelName { get; set; }
 
         /// <summary>
-        /// Gets or sets the <see cref="Type"/> of the model binder used to bind the model.
+        /// Gets or sets the <see cref="Type"/> of the <see cref="IModelBinder"/> implementation used to bind the model.
         /// </summary>
-        public Type BinderType { get; set; }
+        /// <remarks>
+        /// Also set <see cref="BindingSource"/> if the specified <see cref="IModelBinder"/> implementation does not use
+        /// values from form data, route values or the query string.
+        /// </remarks>
+        public Type BinderType
+        {
+            get => _binderType;
+            set
+            {
+                if (!(value == null || typeof(IModelBinder).GetTypeInfo().IsAssignableFrom(value.GetTypeInfo())))
```

Comment:
`if (value != null && !typeof(IModelBinder).IsAssignableFrom(value))`

Reply:
Think I copied the `GetTypeInfo()` use from `BinderTypeModelBinder`. Will fix that up too.